applications/StatisticsApplication/tests/test_spatial_statistics_process_mpi.py

```diff
@@ -0,0 +1,300 @@
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.MetisApplication as KratosMetis
+import KratosMultiphysics.mpi as KratosMPI
+from KratosMultiphysics.process_factory import KratosProcessFactory
+
+import KratosMultiphysics.StatisticsApplication as KratosStats
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+import KratosMultiphysics.kratos_utilities as kratos_utilities
+from KratosMultiphysics.StatisticsApplication.test_utilities import InitializeModelPartVariables
+from KratosMultiphysics.StatisticsApplication.test_utilities import InitializeProcesses
+from KratosMultiphysics.StatisticsApplication.test_utilities import ExecuteProcessFinalizeSolutionStep
+
+
+class SpatialStatisticsProcessMPITest(KratosUnittest.TestCase):
+    def setUp(self):
+        self.model = Kratos.Model()
+        self.model_part = self.model.CreateModelPart("test_model_part")
+        self.model_part.SetBufferSize(1)
+        self.model_part.ProcessInfo.SetValue(Kratos.DOMAIN_SIZE, 2)
+
+        input_filename = "spatial_statistics_process/spatial_statistics_process"
+
+        self.__AddNodalSolutionStepVariables()
+        communicator = Kratos.DataCommunicator.GetDefault()
+        if communicator.Size() == 1:
+            self.skipTest("Test can be run only using more than one mpi process")
+
+        if communicator.Rank() == 0 :
```

Comment:
I would use the distributed_model_part_import_utility
and also partition in memory

check the tests in the mappingapp, I think I found quite good solutions on how to check the same thing in serial and in MPI

I can show you if you want